programs/ssl/ssl_client2.c

```diff
@@ -3302,7 +3308,7 @@ int main( int argc, char *argv[] )
      */
 exit:
 #ifdef MBEDTLS_ERROR_C
-    if( ret != 0 )
+    if( ret != 0 && opt.query_config_mode == DFL_QUERY_CONFIG_MODE )
```

Comment:
This change is probably not needed anymore as in the case of query_config `ret` is expected to be equal to 0, right?